### PR TITLE
Ensure that inline_all_trivial_functions() isn't dead-stripped

### PR DESCRIPTION
**Pull request by steven-johnson**
Can break Li2018 in some build environments.

### CODE REVIEW

File: src/Util.cpp [hunk 1]
```diff
@@ -563,6 +563,18 @@ int get_llvm_version() {
...
+    // isn't used by libHalide, but it is used by the Li2018
+    // autoscheduler plugin, so we need it to be present in the final
+    // libHalide. Unfortunately, build environments that static-link
+    // generators might deadstrip this function, causing Li2018
+    // to fail at load time. This useless call ensures that the function
+    // is kept around; calling it with empty arguments is close to free
+    // and this is only called when load_plugin is called anyway.
+    // (If there is a better, portable way to ensure this function
+    // isn't dead-stripped, then let's do that instead.)
+    (void) Halide::Internal::inline_all_trivial_functions({}, {}, {});
```
Comment (abadams) [inline src/Util.cpp:576]: Is it enough to take the address: internal_assert(&Halide::Internal::inline_all_trivial_functions);

Comment (steven-johnson) [reply src/Util.cpp]: Not in the configuration I'm using.

**Comment by alexreinking** [comment]
I'm confused. I thought we agreed in #5228 that statically linking generators wasn't a supported scenario. We had a pretty long discussion about why this isn't portable and @abadams seemed to think it wasn't something anyone was doing anyway?

This is _definitely_ a unix-only setup. On Windows, when you statically link a library, it is _gone_. There is no dynamic lookup in the library loader there, which is needed for this sort of thing to work with loadable modules. Otherwise you have to use a shared library (on all platforms).

**Comment by abadams** [comment]
I think statically linking to libHalide is a terrible idea, but I guess I should have realized that Google, with its penchant for statically linking the world, is doing it.

Fortunately in that context it's a single system where Steven has full control over the build rules, so it can be made to work.

**Comment by alexreinking** [comment]
> Fortunately in that context it's a single system where Steven has full control over the build rules, so it can be made to work.

Then why not link with `-Wl,--whole-archive ... -Wl,--no-whole-archive` and avoid this PR?

**Comment by alexreinking** [comment]
> Then why not link with `-Wl,--whole-archive ... -Wl,--no-whole-archive` and avoid this PR?

From offline discussion: one build post-processes linked binaries (the generator executables) to strip unused symbols. Not sure what a good workaround would be. Maybe some hack via `-include`?

**Comment by steven-johnson** [comment]
Dropping, I'll figure out another way.

**Pull request closed without merging**